src/NewCommand.php

```diff
@@ -143,6 +143,21 @@ protected function execute(InputInterface $input, OutputInterface $output)
         return $process->getExitCode();
     }
 
+    /**
+     * Return `git config --global init.defaultBranch` if set or default to `main`.
+     *
+     * @return string
+     */
+    protected function defaultBranch($default)
+    {
+        $process = new Process(['git', 'config', '--global', 'init.defaultBranch']);
+        $process->run();
+
+        $output = trim($process->getOutput());
+
+        return $process->isSuccessful() && $output ? $output : $default;
```

Comment:
I'd consider removing the parameter from the signature and simply setting the `main` default here.

Reply:
@driesvints Check it out